threat-model/readme.md

```diff
@@ -0,0 +1,360 @@
+
+
+#### 1.6.1 Assets
+
+1. A partial decryption key used to encrypt the input data.
+2. All the assets of from the first and third parties.
+3. Identity of the other coordinators.
+
+#### 1.6.2 Capabilities
+
+1. Coordinators may compromise the availability of the system by refusing to validate the attestation and refusing to supply their key share.
+
+
+#### 1.6.3 Mitigations
+
+1. Helper party networks with a larger number of coordinators could utilize a threshold key which requires less than all key shares to decrypt the data, preventing an individual coordinator from compromising availability. This comes at the cost of more complicated analysis about the ability of an attacker to control different subsets of that helper party network.
+
+
+### 1.7. Helper party collusion
+
+If enough helper parties collude (beyond the proposal specific subset which an attacker is assumed to control), then none of the properties of the system hold. Such scenarios are outside the threat mode.
+
+However, we do assume that an attacker can always control at least one helper party (i.e., there are no perfectly trusted helper parties.)
+
+
+### 1.8 Cloud Providers for Helper Parties
+
+Helper parties may run either on physical machines owned by directly by the aggregator or (more commonly) subcontract with a cloud provider. We assume that an attacker can control some subset of cloud providers.
+
+
+#### 1.8.1 Assets
+
+1. All the assets of the helper party(ies) utilizing the cloud provider.
+2. All the assets of from the first and third parties.
+3. Identity of the helper parties.
+
+#### 1.8.2 Capabilities
+
+1. Cloud providers have all the capabilities of the helper parties utilizing that cloud provider.
+2. If a sufficient number of helper parties utilize a common cloud provider,
+    a. for aggregator networks, it can reconstruct the client/user assets;
+    b. for coordinator networks, it can reconstruct the decryption key.
+
+
+#### 1.8.3 Mitigations
+
+1. Helper parties networks should utilize sufficiently distinct cloud providers beyond the proposal specific subset which the attacker is assumed to control.
+
+
+### 1.9 Operators of TEEs
+
+As a piece of hardware, TEEs will have an operator with access to the machine. Most commonly, this will be a cloud provider. Depending on the specific hardware, there may be known vulnerabilities in which an attacker who only controls the operator can violate the obliviousness of client/user data. These attacks are outside this threat model, but are likely to inform specific web platform decisions about which instantiations of private computation to support.
+
+#### 1.9.1 Assets
```

Comment:
I wasn't quite sure how to approach this (and I'm unsure if the text immediately above is appropriate.) I'm not sure if we should write this out assuming the operator can actually see inside the TEE, and then call out the hardware as a mitigation? Any suggestions here would certainly be appreciated! (cc @martinthomson @marianapr @palenica)

Reply:
I think that we want to be direct about the threat model that a TEE is developed under.  That is, physical access to the hardware might, with some non-trivial effort, be used to access the active state, which would otherwise be protected.

And we have to consider that the operator has physical access.  This is precisely why some of us find use of a TEE uncomfortable/unacceptable.

I don't think we win by prevaricating on this point.  We win by acknowledging it and tackling it head-on.  That means strict requirements on operational practices by TEE operators, audits, and other such non-technical measures.

Reply:
I'm going to open a new issue, as I'd like to get more input from others. TEEs have multiple different configurations which have different models, and I'm not an expert across them.

Reply:
opened #25